apps/website/src/utils/image-converter.ts

```diff
@@ -0,0 +1,102 @@
+import type { Dispatch, SetStateAction } from "react";
+
+import type { ImageMimeType } from "./files";
+
+async function convertHeicToJpeg(file: File) {
+  const heic2any = (await import("heic2any")).default;
+
+  const blob = await heic2any({
+    blob: file,
+    toType: "image/jpeg",
+    quality: 0.9,
+  });
+  return new File([blob as Blob], file.name.replace(/\.[^.]+$/, ".jpeg"), {
+    type: "image/jpeg",
+  });
+}
+
+async function convertAvifToJpeg(file: File): Promise<File> {
+  return new Promise((resolve, reject) => {
+    const img = new Image();
+    const url = URL.createObjectURL(file);
+
+    img.onload = () => {
+      const canvas = document.createElement("canvas");
+      canvas.width = img.width;
+      canvas.height = img.height;
+      const ctx = canvas.getContext("2d");
+      if (!ctx) {
+        URL.revokeObjectURL(url);
+        reject(new Error("Canvas context not available"));
+        return;
+      }
+      ctx.drawImage(img, 0, 0);
+      canvas.toBlob(
+        (blob) => {
+          URL.revokeObjectURL(url);
+          if (!blob) {
+            reject(new Error("Conversion to JPEG failed"));
+            return;
+          }
+          resolve(
+            new File([blob], file.name.replace(/\.avif$/i, ".jpeg"), {
+              type: "image/jpeg",
+            }),
+          );
+        },
+        "image/jpeg",
+        0.9,
+      );
+    };
+
+    img.onerror = () => {
+      URL.revokeObjectURL(url);
+      reject(new Error("Failed to load AVIF image"));
+    };
+
+    img.src = url;
+  });
+}
+
+const conversionFunctions: Record<
+  "image/heic" | "image/heif" | "image/avif",
+  (file: File) => Promise<File>
+> = {
+  "image/heic": convertHeicToJpeg,
+  "image/heif": convertHeicToJpeg,
+  "image/avif": convertAvifToJpeg,
+};
+
+function isConvertibleMimeType(
+  mimeType: string,
+): mimeType is keyof typeof conversionFunctions & ImageMimeType {
```

Comment:
Per my other comment, we cannt assert here that mimeType is a member of `ImageMimeType`, there is nothing in the logic that guarantees this now